Task 5

vladimir-arsenov opened the pull request.

vladimir-arsenov committed:
Task 5

lParzivall commented on src/main/java/org/example/client/ApiClient.java:
```diff
@@ -0,0 +1,37 @@
+
+    public Category[] getCategories() {
+        log.info("Calling API to acquire categories...");
+        ResponseEntity<Category[]> response = restTemplate.getForEntity(categoriesUrl, Category[].class);
```
нет обработки ошибок

lParzivall replied:
нет обработки ошибок

lParzivall commented on src/main/java/org/example/configuration/Config.java:
```diff
@@ -0,0 +1,14 @@
+import org.springframework.web.client.RestTemplate;
+
+@Configuration
+public class Config {
```
назвать точнее HttpConfig

lParzivall commented on src/main/java/org/example/configuration/RunAfterStartup.java:
```diff
@@ -0,0 +1,41 @@
+@LogExecutionTime
+@RequiredArgsConstructor
+@Component
+public class RunAfterStartup {
```
ApplicationStartUpListener

lParzivall commented on src/main/java/org/example/configuration/RunAfterStartup.java:
```diff
@@ -0,0 +1,41 @@
+    private final ConcurrentHashMapRepository<String, Location> locationsRepository;
+    private final ApiClient apiClient;
+
+    @LogExecutionTime
```
зачем, если есть над классом?

lParzivall commented on src/main/java/org/example/controller/CategoryController.java:
```diff
@@ -0,0 +1,48 @@
+import org.example.model.Category;
+import org.example.service.CategoryService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
```
https://www.javadude.com/posts/20040522-import-on-demand-is-evil/

lParzivall commented on src/main/java/org/example/controller/CategoryController.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @PostMapping
+    public ResponseEntity<?> addCategory(@RequestBody Category category) {
```
почему тут wildcard?
ResponseEntity можно убрать и возвращать String dto и все что там еще есть, но это по желанию, или оставить, но только без <?>, должно быть однозначно определено

lParzivall commented on src/main/java/org/example/controller/CategoryController.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<?> deleteCategory(@PathVariable Integer id) {
```
void

lParzivall commented on src/main/java/org/example/exception/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,18 @@
+import java.util.NoSuchElementException;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
```
это хорошо 👍

lParzivall commented on src/main/java/org/example/model/Category.java:
```diff
@@ -0,0 +1,10 @@
+import lombok.Data;
+
+@Data
+public class Category {
```
финализировать поля

lParzivall commented on src/main/java/org/example/service/CategoryService.java:
```diff
@@ -0,0 +1,18 @@
+
+import java.util.List;
+
+public interface CategoryService {
```
такие интерфейсы уже не являются бест практис и создают лишний слой абстракции, если видишь, что реализация будет одна, можно его не создавать
разумеется в будущем, при нескольких реализациях маст хев

lParzivall commented on src/main/resources/application.yaml:
```diff
@@ -0,0 +1,4 @@
+api:
+  url:
+    categories: https://kudago.com/public-api/v1.4/place-categories
```
👍

vladimir-arsenov committed:
minor fixes
changed return type of controller methods
added exception handling in ApiClient.java
removed service layer interfaces
